sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/ControlFlowGraph/ControlFlowGraphTest.cs

```diff
@@ -2780,59 +2954,106 @@ public void Cfg_TryCatchFinally_Return_Nested()
 
-            innerTryEndBlock.Should().BeOfType<BranchBlock>();
-            innerTryEndBlock.SuccessorBlocks.Should().OnlyContain(innerCatchBlock /*ex*/, innerFinallyBlock_ForCatch /*no ex*/);
+            innerTryEndBlock.Should().BeOfType<JumpBlock>();
+            innerTryEndBlock.SuccessorBlocks.Should().OnlyContain(innerFinallyBlock_ForReturn);
 
             innerCatchBlock.Should().BeOfType<SimpleBlock>();
             innerCatchBlock.SuccessorBlocks.Should().OnlyContain(innerFinallyBlock_ForCatch);
 
             innerFinallyBlock_ForReturn.Should().BeOfType<SimpleBlock>();
-            innerFinallyBlock_ForReturn.SuccessorBlocks.Should().OnlyContain(finallyBlock_ForReturn);
+            innerFinallyBlock_ForReturn.SuccessorBlocks.Should().OnlyContain(outerFinallyBlock_ForReturn);
 
             innerFinallyBlock_ForCatch.Should().BeOfType<SimpleBlock>();
-            innerFinallyBlock_ForCatch.SuccessorBlocks.Should().OnlyContain(tryEndBlock);
+            innerFinallyBlock_ForCatch.SuccessorBlocks.Should().OnlyContain(outerTryBlock);
 
-            tryEndBlock.Should().BeOfType<BranchBlock>();
-            tryEndBlock.SuccessorBlocks.Should().OnlyContain(catchBlock /*ex*/, finallyBlock_ForCatch /*no ex*/);
+            outerTryBlock.Should().BeOfType<BranchBlock>();
+            outerTryBlock.SuccessorBlocks.Should().OnlyContain(outerCatchBlock /*ex*/, outerFinallyBlock_ForCatch /*no ex*/);
 
-            catchBlock.Should().BeOfType<SimpleBlock>();
-            catchBlock.SuccessorBlocks.Should().OnlyContain(finallyBlock_ForCatch);
+            outerCatchBlock.Should().BeOfType<SimpleBlock>();
+            outerCatchBlock.SuccessorBlocks.Should().OnlyContain(outerFinallyBlock_ForCatch);
 
-            finallyBlock_ForReturn.Should().BeOfType<SimpleBlock>();
-            finallyBlock_ForReturn.SuccessorBlocks.Should().OnlyContain(exit);
+            outerFinallyBlock_ForReturn.Should().BeOfType<SimpleBlock>();
+            outerFinallyBlock_ForReturn.SuccessorBlocks.Should().OnlyContain(exit);
 
-            finallyBlock_ForCatch.Should().BeOfType<SimpleBlock>();
-            finallyBlock_ForCatch.SuccessorBlocks.Should().OnlyContain(afterFinallyBlock);
+            outerFinallyBlock_ForCatch.Should().BeOfType<SimpleBlock>();
+            outerFinallyBlock_ForCatch.SuccessorBlocks.Should().OnlyContain(afterFinallyBlock);
 
             afterFinallyBlock.Should().BeOfType<SimpleBlock>();
             afterFinallyBlock.SuccessorBlocks.Should().OnlyContain(exit);
         }
 
+        [TestMethod]
+        [TestCategory("CFG")]
+        public void Cfg_TryCatch_ReturnVariable_InCatch()
+        {
+            var cfg = Build(@"
+                var number = 5;
+                try
+                {
+                    bar();
+                    return 0;
+                }
+                catch
+                {
+                    return number;
```

Comment:
Additional tests:
* non-direct child `return` e.g.
```
catch
{
  if (retries > 5) { return; }
}
```

* `return` when there is a `finally` block

Reply:
`return` cannot be inside `finally` - https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0157